packages/app/src/templates/components/template-form/template-form.component.tsx

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import React, { useEffect } from "react"
+import { useForm } from "react-hook-form"
+import { defineMessages } from "react-intl"
+import { intl } from "Renderer/utils/intl"
+import { FunctionComponent } from "Renderer/types/function-component.interface"
+import ButtonComponent from "Renderer/components/core/button/button.component"
+import {
+  DisplayStyle,
+  Type,
+} from "Renderer/components/core/button/button.config"
+import Text, {
+  TextDisplayStyle,
+} from "Renderer/components/core/text/text.component"
+import Loader from "Renderer/components/core/loader/loader.component"
+import { LoaderType } from "Renderer/components/core/loader/loader.interface"
+import { TemplateFormProps } from "App/templates/components/template-form/template-form.interface"
+import {
+  Content,
+  TemplateDetailsWrapper,
+  Buttons,
+  Form,
+  TextArea,
+} from "App/templates/components/template-form/template-form.styled"
+import { TemplateFormTestIds } from "App/templates/components/template-form/template-form-ids.enum"
+import { byteLengthValidator } from "App/templates/components/template-form/byte-length.validator"
+
+const messages = defineMessages({
+  editTitle: { id: "module.templates.editTitle" },
+  newTitle: { id: "module.templates.newTitle" },
+  text: { id: "module.templates.text" },
+  cancel: { id: "module.templates.cancelButton" },
+  save: { id: "module.templates.saveButton" },
+})
+
+export const TemplateForm: FunctionComponent<TemplateFormProps> = ({
+  onClose,
+  onSave,
+  error,
+  template,
+  savingPossible = true,
+  saving = false,
+}) => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    setError,
+  } = useForm({
+    defaultValues: template
+      ? template
+      : {
+          text: "",
+        },
+    mode: "onChange",
+  })
+
+  useEffect(() => {
+    if (error) {
```

Comment:
Not 100% sure, but shouldn't we call e.g. `clearErrors` when error is null or undefined? Not forcing though - it is rather my assumption :D 

```
  const {
    register,
    handleSubmit,
    formState: { errors },
    setError,
    clearErrors,
  } = useForm({
    defaultValues: template
      ? template
      : {
          text: "",
        },
    mode: "onChange",
  })

  useEffect(() => {
    if (error) {
      setError("text", {
        type: "manual",
        message: error?.toString(),
      })
    } else {
      clearErrors()
    }
  }, [error])
```

Reply:
@wojcik92michal Maybe your right, I'll check in https://github.com/mudita/mudita-center/pull/750 and provide the required fixes